src/processes/delete-expired-post-process.js

```diff
@@ -24,14 +24,15 @@ const deleteExpiredPostProcess = async (job, done, queueInjection) => {
             expired_posts[doc.author_id] = [doc._id];
           }
         });
+        // eslint-disable-next-line eqeqeq
         if (expired_posts != {}) {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The comparison `expired_posts != {}` will always return true because JavaScript compares objects by reference, not by structure. If you want to check if the object is empty, you should use `Object.keys(expired_posts).length !== 0`.
```diff
-         // eslint-disable-next-line eqeqeq
-         if (expired_posts != {}) {
+         if (Object.keys(expired_posts).length !== 0) {
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->